cache_none option for memoize to deal with concurrency problem

joeportela opened the pull request:
Fixes #135 by allowing the user to specify if their function never returns None.  If so, we can safely assume a None returned from cache.get() means the key does not exist.

joeportela commented:
Also want to consider the defaulting the value to False or making the kwarg 'cache_none' to make things a bit more clear.

gergelypolonkai commented:
Yes, please change it to False by default, so we revert back to 1.7.1 functionality. It was much better for everyone, or so it seems.

joeportela commented:
> Yes, please change it to False by default, so we revert back to 1.7.1 functionality. It was much better for everyone, or so it seems.

Changed the default value to False and also changed the kwarg to 'cache_none'.  I think for a user, the decision to cache None values is a simpler one to make (don't) than deciding if their function may ever return a None value.

gergelypolonkai commented:
Thanks!
